lib/colorls/core.rb

```diff
@@ -381,7 +381,10 @@ def options(content)
         color = @colors[:dir]
         group = :folders
       else
-        key = File.extname(content.name).delete_prefix('.').downcase.to_sym
+        extension = File.extname(content.name).delete_prefix(".")
+        basename = File.basename(content.name, ".*").delete_prefix(".")
+        key = extension.empty? ? basename : extension
+        key = key.downcase.to_sym
```

Comment:
BTW, the `content.name` is already the basename of the path, and since you only use the `basename` if the `extension` is empty, there is no use in calling `File.basename(content.name, ".*")`.

Reply:
Updated